crates/ide_assists/src/handlers/extract_module.rs

```diff
@@ -0,0 +1,660 @@
+
+    println!("{:?}", body.syntax().text());
+
+    return Some(Module {
+        text_range: stmts_range,
+        name: "modname".to_string(),
+        body,
+        leading_ws: "".to_string(),
+    });
+}
+
+impl Module {
+    fn update_refs(
+        &mut self,
+        covering_ele: &SyntaxNode,
+        ctx: &AssistContext,
+        builder: &mut AssistBuilder,
+    ) -> Option<ast::BlockExpr> {
+        let selection_range = ctx.frange.range;
+
+        let mut vis_node_occurences = Vec::new();
+        for child in covering_ele.children() {
+            if selection_range.contains_range(child.text_range()) {
+                match_ast! {
+                    match child {
+                        ast::Struct(it) => {
+                            let node_def = {
+                                let nod = ctx.sema.to_def(&it)?;
+                                Definition::ModuleDef(nod.into())
+                            };
+
+                            self.get_and_process_usages(ctx, builder, node_def);
+                            vis_node_occurences.push(it.syntax().clone());
+
+                            for desc in it.syntax().descendants() {
+                                if let Some(record_field) = ast::RecordField::cast(desc) {
+                                    vis_node_occurences.push(record_field.syntax().clone());
+                                }
+                            }
+                        },
+                        ast::Union(it) => {
+                            let node_def = {
+                                let nod = ctx.sema.to_def(&it)?;
+                                Definition::ModuleDef(nod.into())
+                            };
+
+                            self.get_and_process_usages(ctx, builder, node_def);
+                            vis_node_occurences.push(it.syntax().clone());
+
+                            for desc in it.syntax().descendants() {
+                                if let Some(record_field) = ast::RecordField::cast(desc) {
+                                    vis_node_occurences.push(record_field.syntax().clone());
+                                }
+                            }
+                        },
+                        ast::Enum(it) => {
+                            let node_def = {
+                                let nod = ctx.sema.to_def(&it)?;
+                                Definition::ModuleDef(nod.into())
+                            };
+
+                            self.get_and_process_usages(ctx, builder, node_def);
+                            vis_node_occurences.push(it.syntax().clone());
+
+                            for desc in it.syntax().descendants() {
+                                if let Some(record_field) = ast::RecordField::cast(desc) {
+                                    vis_node_occurences.push(record_field.syntax().clone());
+                                }
+                            }
+                        },
```

Comment:
```suggestion
                    match (child.clone()) {
                        ast::Adt(it) => {
                            let node_def = {
                                let nod = ctx.sema.to_def(&it)?;
                                Definition::ModuleDef(nod)
                            };

                            self.get_and_process_usages(ctx, builder, node_def);
                            vis_node_occurences.push(child);

                            for desc in it.syntax().descendants() {
                                if let Some(record_field) = ast::RecordField::cast(desc) {
                                    vis_node_occurences.push(record_field.syntax().clone());
                                }
                            }
                        },
```
I think you can collapse those 3 arms like this